services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -0,0 +1,409 @@
+    logging.info(f"Compute descriptive statistics for {dataset=}, {config=}, {split=}")
+    check_split_exists(dataset=dataset, config=config, split=split)
+
+    config_parquet_and_info_step = "config-parquet-and-info"
+    parquet_and_info_best_response = get_previous_step_or_raise(
+        kinds=[config_parquet_and_info_step],
+        dataset=dataset,
+        config=config,
+    )
+    content_parquet_and_info = parquet_and_info_best_response.response["content"]
+    try:
+        split_parquet_files = [
+            parquet_file
+            for parquet_file in content_parquet_and_info["parquet_files"]
+            if parquet_file["config"] == config and parquet_file["split"] == split
+        ]
+        dataset_info = content_parquet_and_info["dataset_info"]
+    except KeyError as e:
+        raise PreviousStepFormatError(
+            (
+                f"Previous step '{config_parquet_and_info_step}' did not return the expected content: "
+                "'parquet_files' or 'dataset_info'. "
+            ),
+            e,
+        ) from e
+
+    if not split_parquet_files:
+        raise ParquetResponseEmptyError("No parquet files found.")
+    features = dataset_info.get("features")
+    if not features:
+        raise PreviousStepFormatError(
+            f"Previous step '{config_parquet_and_info_step}' did not return the expected content: "
+            "no features found in 'dataset_info'. "
+        )
+
+    split_parquets_size = sum(parquet_file["size"] for parquet_file in split_parquet_files)
+    if split_parquets_size > max_parquet_size_bytes:
+        raise SplitWithTooBigParquetError(
+            f"Statistics computation is limited to split parquets under {max_parquet_size_bytes} bytes. "
+            f"Current size of sum of split parquets is {split_parquets_size} bytes."
+        )
+    parquet_files_urls = [parquet_file["url"] for parquet_file in split_parquet_files]
+
+    stats: List[StatisticsPerColumnItem] = []
+    num_examples = dataset_info["splits"][split]["num_examples"]
+    categorical_features = {
+        feature_name: feature for feature_name, feature in features.items() if feature.get("_type") == "ClassLabel"
+    }
+    numerical_features = {
+        feature_name: feature
+        for feature_name, feature in features.items()
+        if feature.get("_type") == "Value" and feature.get("dtype") in NUMERICAL_DTYPES
+    }
+    if not categorical_features and not numerical_features:
+        raise NoSupportedFeaturesError(
+            "No columns for statistics computation found. Currently supported feature types are: "
+            f"{NUMERICAL_DTYPES} and ClassLabel. "
+        )
+
+    con = duckdb.connect(":memory:")  # we don't load data in local db file, use local parquet file instead
+    # configure duckdb extensions
+    con.sql(f"SET extension_directory='{local_parquet_directory}';")
+    con.sql("INSTALL httpfs")
+    con.sql("LOAD httpfs")
+    con.sql("SET enable_progress_bar=true;")
```

Comment:
does it generate logs? with which level? not important now, but we might have to monitor to avoid flooding the production logs